contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,143 @@
+        ///            
+        /// 
+        /// Parameters: capability: The capability to expose a certain fungible
+        /// token vault deposit function through {FungibleToken.Receiver} that
+        /// will be added to the switchboard
+        ///
+        pub fun addNewVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // to store inside the switchboard
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference type as key for storing the capability
+            self.receiverCapabilities[vaultRef.getType()] = capability
+            // Emit the event that indicates that a new capability has been added
+            emit VaultCapabilityAdded(type: vaultRef.getType())
+        }
+
+        /// removeVault removes a fungible token receiver capability 
+        ///                       from the switchboard resource
+        /// 
+        /// Parameters: capability: The capability to a fungible token vault 
+        ///                         to be removed from the switchboard
+        ///
+        pub fun removeVault(capability: Capability<&{FungibleToken.Receiver}>) {
+            // Borrow a reference to the vault pointed by the capability we want
+            // store inside the switchboard            
+            let vaultRef = capability.borrow() 
+                ?? panic ("Cannot borrow reference to vault from capability")
+            // Use the vault reference to find the capability to remove
+            self.receiverCapabilities.remove(key: vaultRef.getType())
+            // Emit the event that indicates that a new capability has been removed
+            emit VaultCapabilityRemoved(type: vaultRef.getType())            
+        }
+        
+        /// deposit Takes a fungible token vault and routes it to the proper
+        ///         fungible token receiver capability for depositing it
+        /// 
+        /// Parameters: from: The deposited fungible token vault resource
+        ///        
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let depositedVaultCapability = self.receiverCapabilities[from.getType()] ?? 
+                panic ("The deposited vault is not available on this switchboard")
+            let vaultRef = depositedVaultCapability.borrow() ?? 
+                panic ("Can not borrow a reference to the the vault")
+            vaultRef.deposit(from: <- from)
+        }
+
+        /// getVaultCabilities function for get to know which tokens a certain
+        /// switchboard resource is prepared to receive
+        ///
+        /// Returns: The dictionary of stored {FungibleToken.Receiver} 
+        /// capabilities
+        ///
+        pub fun getVaultCapabilities(): {Type: Capability<&{FungibleToken.Receiver}>} {
+            return self.receiverCapabilities
+        }
```

Comment:
Do we need to give out Capabilities as a map ? I feel like just returning `keys` from here will be more useful. 

This would remove need for a public receiver link also.

Reply:
yeah, returning just the keys would be better

Reply:
great I'll change the function so it returns the keys, thanks @bluesign! Do you think getVaultTypes would be an ok name for the function? 

But I don't really understand why this change will remove the need for the public capability, could you help me understanding it?

Reply:
People can add `private` capability, and then it can leak. But with this way, capability never leaves the resource.

Reply:
I pushed the commit for this change, @satyamakgec could you take a look at the test for it? I'm checking for the result not to be null `expect(result).not.toBe(null);` but I don't like it, I'd love to have a test that checks for a non empty array but I didn't find a way to do that